### ISSUE TITLE
Model, created in itself static method, doesn't work properly

### ISSUE THREAD

**Issue by Amareis** [xaviergonz/mobx-keystone#173]
https://codesandbox.io/s/mobx-keystone-static-method-problem-4thb2?file=/src/index.tsx

```ts
import { Model, tProp, model } from "mobx-keystone";

@model("BuggyModel")
class BuggyModel extends Model({
  value: tProp("1")
}) {
  static foo() {
    return new BuggyModel({});
  }
}

const a = BuggyModel.foo();
a.value = null as any; //don't throws!
console.log(a); //there is no modelInitializedSymbol in instance

const b = new BuggyModel({})
b.value = null as any; //throws, ok
```

I think it's coupled with babel, because I cannot reproduce it in pure ts tests. Possible related #18 #92

**Comment by xaviergonz** [comment]
I think it is this babel decorator plugin bug: https://github.com/babel/babel/issues/11131

**Comment by xaviergonz** [comment]
Also related: https://github.com/xaviergonz/mobx-keystone/issues/122

**Comment by Amareis** [comment]
Yep, definitely it's babel bug. Currently i use external method and assign it into static property
```ts
static foo = foo
```

Can we fix this babel bug? I'll try to look into this.

**Comment by xaviergonz** [comment]
If you were to fix the babel plugin that'd be awesome :) Not much that can be done on the mobx-keystone repo sadly

**Comment by Amareis** [comment]
Uh-oh, so I fixed plugin, but not pushed it yet.

**Comment by Amareis** [comment]
https://github.com/babel/babel/pull/12007

**Comment by xaviergonz** [comment]
That's awesome! Let's hope it gets merged soon 👍

**Comment by vojto** [comment]
Workarounds in that issue aren't very appealing. I think we'll just avoid static methods for now.

**Comment by samskiter** [comment]
womp womp - is this still an issue?

**Comment by xaviergonz** [comment]
It never got fixed by babel, so yes. I don't know if it happens with the new (non legacy) decorators though.

**Comment by alex-chuev** [comment]
This killed my whole work day!

**Comment by elliotmoose** [comment]
> With `experimentalDecorators: true`, TypeScript compiles self-references in decorated classes using an internal `_1` variable that points to the **pre-decoration** class, bypassing the `@model` Proxy. This means `runAfterClassInitialization` never fires, `modelInitializedSymbol` stays `false`, and all property setters silently no-op.

Summary from opus. I think the biggest challenge was how difficult it was to pinpoint this issue, no errors no nothing.
